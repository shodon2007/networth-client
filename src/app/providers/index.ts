import {AppRouter} from "./router";
import {setTheme} from "./themeProvider/model/slice/themeSlice";

export {AppRouter, setTheme};
