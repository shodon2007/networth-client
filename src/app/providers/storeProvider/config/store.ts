import {configureStore} from "@reduxjs/toolkit";
import {rootReducer} from "src/shared/lib/store/config/reducer";
import {userApi} from "src/entities";
import {setupListeners} from "@reduxjs/toolkit/query";
import {profileApi} from "src/entities/Profile";

const store = configureStore({
	reducer: rootReducer,
	middleware: (getDefaultMiddleware) => {
		return getDefaultMiddleware()
			.concat(userApi.middleware)
			.concat(profileApi.middleware);
	},
});
setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
