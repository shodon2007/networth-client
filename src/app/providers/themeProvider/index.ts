import {getTheme} from "./model/selectors/getTheme";
import {themeReducer} from "./model/slice/themeSlice";
export {themeReducer, getTheme};
