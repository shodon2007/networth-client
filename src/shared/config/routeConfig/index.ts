export {AppRoutes, routePath, routes} from "./routeConfig";
export type {RouteItem} from "./routeConfig";
