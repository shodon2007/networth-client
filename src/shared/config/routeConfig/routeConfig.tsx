import {ReactNode} from "react";
import {ActivationPageAsync} from "src/pages/Auth/Activation";
import {LoginPageAsync} from "src/pages/Auth/Login";
import {RegistrationPageAsync} from "src/pages/Auth/Registration";
import {FeedPageAsync} from "src/pages/Feed";
import {MessengerPageAsync} from "src/pages/Messenger";
import {ProfilePageAsync} from "src/pages/Profile";
import {Layout} from "src/shared/ui/Layout/Layout";

export enum AppRoutes {
	REGISTRATION = "registration",
	LOGIN = "login",
	ACTIVATION = "activation",

	HOME = "home",
	PROFILE = "profile",
	MESSENGER = "messenger",
}

export const routePath: Record<AppRoutes, string> = {
	[AppRoutes.HOME]: "",
	[AppRoutes.PROFILE]: "/profile",
	[AppRoutes.LOGIN]: "/login",
	[AppRoutes.REGISTRATION]: "/registration",
	[AppRoutes.ACTIVATION]: "/activation",
	[AppRoutes.MESSENGER]: "/messanger",
};

export interface RouteItem {
	path: string;
	element: ReactNode;
	index?: boolean;
	children?: RouteItem[];
}

export const routes: RouteItem[] = [
	{
		path: routePath.home,
		element: <Layout />,
		children: [
			{
				path: routePath.home,
				element: <FeedPageAsync />,
				index: true,
			},
			{
				path: routePath.profile,
				element: <ProfilePageAsync />,
				index: true,
			},
			{
				path: routePath.messenger,
				element: <MessengerPageAsync />,
				index: true,
			},
			{
				path: routePath.activation,
				element: <ActivationPageAsync />,
			},
			{
				path: routePath.registration,
				element: <RegistrationPageAsync />,
			},
			{
				path: routePath.login,
				element: <LoginPageAsync />,
			},
		],
	},
];
