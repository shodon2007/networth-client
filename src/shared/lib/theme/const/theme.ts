export enum Themes {
    DARK = "app_global_dark_theme",
    LIGHT = "app_global_light_theme",
}
