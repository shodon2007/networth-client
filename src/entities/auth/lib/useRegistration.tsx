import {useMutation} from "@tanstack/react-query";
import {toast} from "react-toastify";
import {useNavigate} from "react-router-dom";
import {AxiosError} from "axios";

import {useAppDispatch} from "src/shared/lib/store";
import {setUser} from "src/entities/User/services/userSlice";
import {ErrorResponseType} from "src/shared/types/response/responseType";
import {RegistrationRequestType, getRegistration} from "../api/authApi";

export const useRegistration = () => {
	const dispatch = useAppDispatch();
	const navigate = useNavigate();

	return useMutation({
		mutationFn: ({email, password, name, surname}: RegistrationRequestType) => {
			return getRegistration({email, password, name, surname});
		},
		onSuccess: (resp) => {
			dispatch(setUser(resp.data.data));
			toast.success(resp.data.message);
			navigate("/activation");
		},
		onError: (resp: AxiosError<ErrorResponseType>) => {
			toast.error(resp.response?.data.message);
		},
	});
};
