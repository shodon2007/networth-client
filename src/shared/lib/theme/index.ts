import {Themes} from "./const/theme";
import {useTheme} from "./hooks/useTheme";

export {Themes, useTheme};
