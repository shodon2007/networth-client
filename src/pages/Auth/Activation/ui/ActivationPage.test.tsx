import {waitFor} from "@testing-library/dom";
import {RootState} from "src/app/providers/storeProvider";
import {routePath} from "src/shared/config/routeConfig";
import {Themes} from "src/shared/lib/theme";
import {renderWithProviders} from "src/shared/tests";

const mockAuthStore: RootState = {
	theme: {
		theme: Themes.DARK,
	},
	user: {
		isUserDataLoading: false,
		accessToken: "меега пачка чипсонов",
		refreshToken: "со вкусом краба",
		data: {
			email: "mockEmail2007@youtube.com",
			id: 1,
			isActivated: 0,
			name: "shodon",
			phoneNumber: "321313",
			surname: "shodon",
			avatar: "https://avatars.githubusercontent.com/u/133156395?v=4",
		},
		isAuth: true,
	},
};

describe("Testing ActivationPage.tsx", () => {
	it("should match the snapshot", async () => {
		const screen = renderWithProviders(routePath.activation, mockAuthStore);

		expect(await screen.findByTestId("activation-page")).toBeInTheDocument();
		expect(await screen.findByTestId("activation-page")).toMatchSnapshot();
	});
	it("should match the mock email", async () => {
		const screen = renderWithProviders(routePath.activation, mockAuthStore);

		expect(await screen.findByTestId("activation-page")).toBeInTheDocument();
		expect(screen.getByTestId("activation-email")).toContainHTML(
			"mockEmail2007@youtube.com",
		);
	});
	it("should match the mock email", async () => {
		const screen = renderWithProviders(routePath.activation, mockAuthStore);

		expect(await screen.findByTestId("activation-page")).toBeInTheDocument();
		expect(screen.getByTestId("activation-email")).toContainHTML(
			"mockEmail2007@youtube.com",
		);
	});
	it("should navigate to login page when user is not auth", async () => {
		const screen = renderWithProviders(routePath.activation, {
			user: null,
		});

		await waitFor(() =>
			expect(screen.queryByTestId("activation-page")).not.toBeInTheDocument(),
		);
		await waitFor(() =>
			expect(screen.getByTestId("login-form")).toBeInTheDocument(),
		);
	});
	it("should navigate to home page when user is not auth", async () => {
		const newMockStore = JSON.parse(JSON.stringify(mockAuthStore));
		newMockStore.user.data.isActivated = 1;
		const screen = renderWithProviders(routePath.activation, newMockStore);

		await waitFor(() =>
			expect(screen.queryByTestId("activation-page")).not.toBeInTheDocument(),
		);
		await waitFor(() =>
			expect(screen.getByTestId("home-page")).toBeInTheDocument(),
		);
	});
});
