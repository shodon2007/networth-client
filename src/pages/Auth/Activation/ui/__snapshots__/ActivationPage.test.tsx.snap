// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ActivationPage.tsx should match the snapshot 1`] = `
<div
  data-testid="activation-page"
>
  <div
    class="block default MainBlock"
    data-testid="block"
  >
    <h1>
      Hello
    </h1>
    <span
      class="EmailNotification"
    >
      На почту
       
      <span
        data-testid="activation-email"
        style="font-weight: bold;"
      >
        mockEmail2007@youtube.com
      </span>
       
      отправлена ссылка для подтветждения
    </span>
    <ul
      class="LinksList"
    >
      <a
        aria-current="page"
        class="link active"
        href="/activation"
      >
        Home
      </a>
      <a
        class="link"
        href="/profile"
      >
        Profile
      </a>
    </ul>
  </div>
</div>
`;
